src/components/HeartRateZoneGraph.vue

```diff
@@ -381,29 +392,41 @@ export default {
             // calculate the bpm step
             const zonesInvolved = this.determineZonesInvolved(hrZoneIndex, nextHrZoneIndex)
             const totalSteps = zonesInvolved.reduce((total, zoneIndex) => {
-              const start = Math.min(
-                Math.max(this.hrZones[zoneIndex].minmax[0], entry.heartRate || 0),
-                this.hrZones[zoneIndex].minmax[1]
-              )
-              const end = Math.min(
-                Math.max(this.hrZones[zoneIndex].minmax[0], nextEntry.heartRate || 0),
-                this.hrZones[zoneIndex].minmax[1]
-              )
+              const start =
+                zoneIndex == -1
+                  ? entry.heartRate ?? 0
+                  : Math.min(
+                      Math.max(this.hrZones[zoneIndex].minmax[0], entry.heartRate ?? 0),
+                      this.hrZones[zoneIndex].minmax[1]
+                    )
+              const end =
+                zoneIndex == -1
+                  ? nextEntry.heartRate ?? 0
+                  : Math.min(
+                      Math.max(this.hrZones[zoneIndex].minmax[0], nextEntry.heartRate ?? 0),
+                      this.hrZones[zoneIndex].minmax[1]
+                    )
               return 1 + total + (Math.max(end, start) - Math.min(end, start))
```

Comment:
this can be simplified though

```ts
return 1 + total + Math.abs(start-end)
```

btw one question pls, if `zoneIndex == -1` what will happen then? Can you give a little illustration regarding this?

Reply:
entry.heartRate = 50
nextEntry.heartRate = 200

assume 50 is below zone 1, and 200 is above zone 5, it will belong to the zone who has (200-50) 150bpm in it?

Reply:
maybe more real world case:

entry.heartRate = 50 (assume not belong to zone)
entry.heartRate = 90 (assume zone 1)

it will add 40 steps then distribute them evenly based on elapsed?

Reply:
```
entry.heartRate = 50
nextEntry.heartRate = 200
```

It will do transition 50 (inclusive) to 200 (inclusive) with each value added time to zone `(1/151 * deltaTime)`
roughly figure like this
```
200 BPM = 1/151 * deltaTime -> Zone 5
...
50 BPM = 1/151 * deltaTime -> No Zone
```

because Zone 5 is 90% to Infinity, therefore 200+ still counted as Zone 5.
because Zone 1 is 50%-60%, therefore 50 didn't counted as Zone 1 and discarded.